c++/cpp2py/converters/optional.hpp

```diff
@@ -3,12 +3,12 @@ namespace cpp2py {
 
  template <typename T> struct py_converter<std::optional<T>> {
 
-  using conv = py_converter<T>;
+   using conv = py_converter<std::decay_t<T>>;
 
-  static PyObject *c2py(std::optional<T> &op) {
-   if (!bool(op)) Py_RETURN_NONE;
-   return conv::c2py(*op);
-  }
+   static PyObject *c2py(std::optional<T> op) {
+     if (!bool(op)) Py_RETURN_NONE;
+     return conv::c2py(std::move(*op));
```

Comment:
Isn't rather 
std::move(op).value() or  *(std::move(op)) [ the second is somewhat less clear]
you have to move the op itself I think.. Not sure.

Reply:
There should be no problem with the syntax.
```c++
std::move(*op)
```
We invoke
```c++
T& operator*() &;
```
and then move from the `T&`.